sklearn/preprocessing/_discretization.py

```diff
@@ -233,23 +246,28 @@ def fit(self, X, y=None):
                 '`subsample` must be used with `strategy="quantile"`.'
             )
 
+        elif self.strategy != "quantile" and sample_weight is not None:
+            raise ValueError(
+                "`sample_weight` was provided but it can be only used with"
+                f"strategy='quantile'. Got strategy={self.strategy!r} instead."
+            )
+
         valid_encode = ("onehot", "onehot-dense", "ordinal")
         if self.encode not in valid_encode:
             raise ValueError(
                 "Valid options for 'encode' are {}. Got encode={!r} instead.".format(
                     valid_encode, self.encode
                 )
             )
-        valid_strategy = ("uniform", "quantile", "kmeans")
-        if self.strategy not in valid_strategy:
-            raise ValueError(
-                "Valid options for 'strategy' are {}. "
-                "Got strategy={!r} instead.".format(valid_strategy, self.strategy)
-            )
 
         n_features = X.shape[1]
         n_bins = self._validate_n_bins(n_features)
 
+        if sample_weight is not None:
+            sample_weight = np.array(
+                _check_sample_weight(sample_weight, X, dtype=X.dtype, copy=True)
+            )
```

Comment:
This should already be a numpy array, right?

```suggestion
            sample_weight = _check_sample_weight(
                sample_weight, X, dtype=X.dtype, copy=True
            )
```

Reply:
I corrected this in https://github.com/scikit-learn/scikit-learn/pull/22048/commits/30edabcbd6efb87d38bfb47f004ca36a7cbea664 (l.267)